compose-material/src/main/java/com/google/android/horologist/compose/material/Title.kt

```
@@ -50,14 +70,53 @@ public fun Title(
 @Composable
 public fun Title(
     text: String,
-    modifier: Modifier = Modifier
+    modifier: Modifier = Modifier,
+    textType: TextType = TextType.Primary,
+    icon: ImageVector? = null,
+    iconTint: Color = MaterialTheme.colors.onBackground,
+    iconSize: Dp = 24.dp
 ) {
-    Text(
-        text = text,
-        modifier = modifier.semantics { heading() },
-        textAlign = TextAlign.Center,
-        overflow = TextOverflow.Ellipsis,
-        maxLines = 3,
-        style = MaterialTheme.typography.title3
-    )
+    Row(
+        modifier = Modifier
+            .padding(horizontal = 14.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        icon?.let {
```

Icon should only be displayed for secondary title, so it should add a check here to ignore the param or throw an exception to fail fast and let the developer know the proper usage. 

That might be a confusing API, so I'm leaning towards having a separate `Title` function for the secondary title. But if we use the same name, the signatures might be the same and it will be difficult to use the secondary title:

```kotlin
fun Title(text: String, modifier: Modifier, ...) {} // primary
fun Title(text: String, modifier: Modifier, icon: ImageVector? = null, ...) {} // secondary

// this might call the primary title:
Title(text = "I want a secondary title with no icon")
```

So maybe this extra function should be called `SecondaryTitle`. Thoughts?

---

Thanks, that's a good observation.
Agreed and added the 'SecondaryTitle' function for simplicity :)